### PR TITLE
Implement devise

### PR DESCRIPTION
**Pull request by lordksix**
In this PR, the changes are as follows:
- [x] Remove current_user method in ApplicationController, devise will provide us with one.
  - [x] Install and setup devise
  - [x] Can register a new user.
  - [x] User logs in with a combination of email and password.
  - [x] Hashed password should be stored in the database.
  - [x] Ask for confirmation of email.
  - [x] Can reset password.
- [x] Modify devise views

### CODE REVIEW

File: app/views/shared/_nav_menu_header.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<header class="w-full p-4">
+  <div class="flex flex-wrap items-center justify-end gap-4 w-10/12 mx-auto">
+    <% if user_signed_in? %>
+      <%= link_to "Sign out", destroy_user_session_path, class: 'px-4 py-2 border-2 border-black cursor:pointer', method: :delete %>
```
Comment (FrederickMih) [inline app/views/shared/_nav_menu_header.html.erb]: - [x] I got the error on the screenshot below after clicking on this link to sign out. Let's kindly fix it.
Screenshot from 2023-08-31 01-37-36

File: app/models/user.rb [hunk 2]
```diff
@@ -1,4 +1,8 @@
 class User < ApplicationRecord
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :validatable
```
Comment (FrederickMih) [inline app/models/user.rb]: - [x] The application should ask for email confirmation. Kindly enable `:confirmation` here and make sure a user can confirm his email

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand: :raising_hand: @lordksix, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!





## Highlights :green_circle: 

- Login and registration views have been modified :+1: 
- App allow user to successfully register :+1: 


## Required Changes ♻️
**N/A**
_Check the comments under the review._

## Optional suggestions
**N/A**
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/user.rb (hunk 2 above)
Comment (lordksix) [reply app/models/user.rb]: Thank you for the feedback. It has been added.

File: app/views/shared/_nav_menu_header.html.erb (hunk 1 above)
Comment (lordksix) [reply app/views/shared/_nav_menu_header.html.erb]: Thank you for the feedback. The issues has been resolved

**Review by OLIPLICHE — APPROVED** [review]
Hi @lordksix, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Applied the changes requested((✅)) 👏
- A user can log in with a combination of email and password 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**